xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -375,61 +383,23 @@ public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
         }
 
         LoadStatsStore loadStatsStore = new LoadStatsStoreImpl();
+        addLoadStatsStore(clusterServiceName, loadStatsStore);
         localityStore = localityStoreFactory.newLocalityStore(helper, lbRegistry, loadStatsStore);
-        LoadReportCallback lrsCallback =
-            new LoadReportCallback() {
-              @Override
-              public void onReportResponse(long reportIntervalNano) {
-                localityStore.updateOobMetricsReportInterval(reportIntervalNano);
-              }
-            };
-
-        // TODO(zdapeng): Use XdsClient to do Lrs directly.
-        // For now create an LRS Client.
-        if (channel != null) {
-          lrsClient =
-              loadReportClientFactory.createLoadReportClient(
-                  channel,
-                  helper.getAuthority(),
-                  Node.getDefaultInstance(),
-                  helper.getSynchronizationContext(),
-                  helper.getScheduledExecutorService(),
-                  new ExponentialBackoffPolicy.Provider(),
-                  GrpcUtil.STOPWATCH_SUPPLIER);
-          lrsClient.addLoadStatsStore(edsServiceName, loadStatsStore);
-        } else {
-          lrsClient = new LoadReportClient() {
-            @Override
-            public void startLoadReporting(LoadReportCallback callback) {}
-
-            @Override
-            public void stopLoadReporting() {}
-
-            @Override
-            public void addLoadStatsStore(
-                String clusterServiceName, LoadStatsStore loadStatsStore) {
-            }
-
-            @Override
-            public void removeLoadStatsStore(String clusterServiceName) {
-            }
-          };
-        }
 
-        endpointWatcher = new EndpointWatcherImpl(lrsClient, lrsCallback, localityStore);
-        xdsClient.watchEndpointData(edsServiceName, endpointWatcher);
-        if (oldEndpointWatcher != null && oldEdsServiceName != null) {
-          xdsClient.cancelEndpointDataWatch(oldEdsServiceName, oldEndpointWatcher);
+        endpointWatcher = new EndpointWatcherImpl(localityStore);
+        xdsClient.watchEndpointData(clusterServiceName, endpointWatcher);
+        if (oldEndpointWatcher != null && oldClusterServiceName != null) {
+          xdsClient.cancelEndpointDataWatch(oldClusterServiceName, oldEndpointWatcher);
         }
         LookasideLb.this.endpointWatcher = endpointWatcher;
```

Comment:
>However, the field for the inner class is not needed after not overriding `handleResolvedAddresses()`.

I take it back, the field for the inner class is also needed because `ClusterEndpointsBalancer` owns the lifecycle of the watcher: `ClusterEndpointsBalancer.shutdown()` cancels the watcher.